cmd/argo/commands/server.go

```diff
@@ -105,11 +108,30 @@ See %s`, help.ArgoServer),
 
 			var tlsConfig *tls.Config
 			if secure {
-				log.Infof("Generating Self Signed TLS Certificates for Secure Mode")
-				tlsConfig, err = tlsutils.GenerateTLSConfig()
+				tlsMinVersion := env.GetInt("TLS_MIN_VERSION", tls.VersionTLS12)
```

Comment:
I'm pretty sure there is existing func for this. Adding a new go.mod dependency increases our vulnerability surface, so please remove this.